src/com/carolinarollergirls/scoreboard/defaults/DefaultSkaterModel.java

```diff
@@ -151,6 +152,7 @@ public void AddPenaltyModel(String id, boolean foulout_explusion, int period, in
         if (penalties.size() < 9) {
           DefaultPenaltyModel dpm = new DefaultPenaltyModel(id, period, jam, code);
           penalties.add(dpm);
+          sortPenalties();
```

Comment:
The EVENT_PENALTY is no longer just about that individual penalty, as it could reorder other penalties. Thus I think the event should be considered to be more generic and not pass dpm to make that clearer. This doesn't require code changes in xml/json listeners as they currently replace the entire skater.

Reply:
I agree that a single penalty object is not ideal, but I also want to avoid the potential pitfalls of not sending anything at all in as the event value (as that would violate the convention that null is a removal action of some sort).  For now I've updated it to be the skater's list of penalties, but I can update it to be something else (like total count of penalties, a string/enum indicating the action type, etc) if you think that's more appropriate.

Reply:
sgtm